### PR TITLE
[Feature] Support Hive timestamp partition column

### PR DESCRIPTION
**Pull request by Smith-Cruise**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/analysis/DateLiteral.java [hunk 1]
```diff
@@ -260,11 +260,22 @@ public String getStringValue() {
     private String convertToString(PrimitiveType type) {
         if (type == PrimitiveType.DATE) {
             return String.format("%04d-%02d-%02d", year, month, day);
-        } else if (microsecond == 0) {
-            return String.format("%04d-%02d-%02d %02d:%02d:%02d", year, month, day, hour, minute, second);
         } else {
-            return String.format("%04d-%02d-%02d %02d:%02d:%02d.%06d", year, month, day, hour, minute, second,
-                    microsecond);
+            if (hiveFormat) {
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/analysis/DateLiteral.java:264]:
```
else if (hiveFormat && microsecond == 0) {
    return String.format("%04d-%02d-%02d %02d:%02d:%02d.0", year, month, day, hour, minute, second);
} else if (microsecond == 0) {
    return String.format("%04d-%02d-%02d %02d:%02d:%02d", year, month, day, hour, minute, second);
}  else {
    return String.format("%04d-%02d-%02d %02d:%02d:%02d.%06d", year, month, day, hour, minute, second,
                            microsecond);
}
```

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 11 / 19 (57.89%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PruneHDFSScanColumnRule.java|1|2|50.00%|[153]|
|:large_blue_circle:|com/starrocks/analysis/DateLiteral.java|6|12|50.00%|[265, 266, 268, 269, 440, 441]|
|:large_blue_circle:|com/starrocks/connector/PartitionUtil.java|4|5|80.00%|[124]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 18 / 19 (94.74%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PruneHDFSScanColumnRule.java|1|2|50.00%|[153]|
|:large_blue_circle:|com/starrocks/analysis/DateLiteral.java|12|12|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/PartitionUtil.java|5|5|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 11 / 12 (91.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PruneHDFSScanColumnRule.java|1|2|50.00%|[153]|
|:large_blue_circle:|com/starrocks/analysis/DateLiteral.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/PartitionUtil.java|3|3|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 13 / 14 (92.86%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PruneHDFSScanColumnRule.java|1|2|50.00%|[153]|
|:large_blue_circle:|com/starrocks/analysis/DateLiteral.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/PartitionUtil.java|3|3|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 10 / 11 (90.91%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PruneHDFSScanColumnRule.java|1|2|50.00%|[153]|
|:large_blue_circle:|com/starrocks/analysis/DateLiteral.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/PartitionUtil.java|3|3|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 10 / 11 (90.91%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PruneHDFSScanColumnRule.java|1|2|50.00%|[153]|
|:large_blue_circle:|com/starrocks/analysis/DateLiteral.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/PartitionUtil.java|3|3|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 12 / 18 (66.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PruneHDFSScanColumnRule.java|1|2|50.00%|[153]|
|:large_blue_circle:|com/starrocks/connector/PartitionUtil.java|11|16|68.75%|[204, 205, 207, 209, 210]|

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 11 / 12 (91.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PruneHDFSScanColumnRule.java|1|2|50.00%|[153]|
|:large_blue_circle:|com/starrocks/connector/PartitionUtil.java|10|10|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 11 / 12 (91.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PruneHDFSScanColumnRule.java|1|2|50.00%|[153]|
|:large_blue_circle:|com/starrocks/connector/PartitionUtil.java|10|10|100.00%|[]|

**Review by dirtysalt — APPROVED** [review]

**Comment by Smith-Cruise** [comment]
@Mergify backport branch-3.1

**Pull request merged**